### ISSUE TITLE
Push participations + retractions to the relay system

### ISSUE THREAD

**Issue by jaywink** [diaspora/diaspora#7123]
Currently since 0.6 podmins can push public posts out to [the relay system](https://github.com/jaywink/social-relay/blob/master/docs/relays.md). This setting is off by default. Currently there are two active relays ([1](https://relay.iliketoast.net/), [2](https://relay.diasp.org/)) known to me which in the last week have delivered almost 12K public posts or participations on public posts to ~40 distinct pods.

Diaspora does not yet support pushing participations or retractions to the relay system, only posts. Participations and retractions will be delivered by the relays to the same destination where the originating post was delivered to. This allows pods which are fully disconnected from each other by social interaction to receive not only the posts but also the participations (solving or at least helping #2407).

Diaspora should, if relay outbound setting is enabled, send also participations (comments, likes, participations) and retractions (for posts) to the relay in the settings. See [relay document](https://github.com/jaywink/social-relay/blob/master/docs/relays.md#participations-comments-likes-and-participations).

Basically TLDR, just like status message, a simple CC of the payload to the relay.

**Comment by jaywink** [comment]
Just tested that the relay server now relays also post retractions, according to the same rule as for participations, ie to the targets where the post was also delivered to.

**Comment by cmrd-senya** [comment]
> Just tested that the relay server now relays also post retractions,
according to the same rule as for participations, ie to the targets
where the post was also delivered to.

That is what actually diaspora should do itself: send retractions to
those where post went before, not only to current contacts and participants.

**Comment by jaywink** [comment]
@cmrd-senya the problem is that the posts, comments and likes might be delivered by the relay system to totally disconnected pods that have absolutely no social route to the place where the retraction happens. In this case, the retraction must be delivered via the same channel because the server itself cannot know where to delivery it.

Unless there is some kind of "ping I received this" kind of mechanism in diaspora which I am not aware of. By this I mean that if my fully disconnected pod receives a post, it would ping back to the source *and listen for events on the original content, even though no user is directly participating*.

If there are reasons not to carbon copy (opt-in) to the relays also comments, likes and retractions. sure I would very much like to hear them. There are a lot of posts already moving through the relays (800-1500 per day on mine). I'm hesitant to "market" the system more to podmins until the participations and retractions is implemented in at least diaspora. I've requested also Friendica and Hubzilla to add the same, which have had posts relay support before 0.6 already.

**Comment by jaywink** [comment]
As pointed out [in discourse](https://discourse.diasporafoundation.org/t/public-post-federation/264/123), diaspora has no intention to send interactions to the relay. Instead, all received posts will be auto-subscribed to, which will cause source nodes to deliver interactions themselves.

Closing.
